### PR TITLE
Add Objectable behavior 

### PR DESCRIPTION
**Pull request by bahaagalal**

### CODE REVIEW

File: src/behaviors.js [hunk 1]
```diff
@@ -108,6 +109,38 @@ class Versionable {
     }
 }
 
+class Objectable {
```
Comment (a-saad) [inline src/behaviors.js]: I propose ObjectIDable.

File: src/behaviors.js [hunk 2]
```diff
@@ -108,6 +109,38 @@ class Versionable {
...
+        this.mongoObjectIdFields = (config && config.mongoObjectIdFields)? config.mongoObjectIdFields : ['_id'];
+    }
+
+    _convertToObjectId(doc) {
+        this.mongoObjectIdFields.forEach((field) => {
+            if(doc[field]) doc[field] = new ObjectID(doc[field].toString());
+        });
+    }
+
+    beforeUpdateOne(filter, update, options){
```
Comment (a-saad) [inline src/behaviors.js]: Add beforeInsert* observers. The ObjectID may also be manually generated.

File: test/behaviorsSpec.js [hunk 3]
```diff
@@ -370,6 +371,157 @@ describe('Behaviors', () => {
...
+				let result = objectable.beforeUpdateOne({
+					'_id': objectId
+				}, {
+					$set: {
+						'name': 'crowdanalyzer'
+					}
+				});
+				expect(result).to.be.an('Array');
+				expect(result[0]).to.be.an('object');
+				expect(result[0]._id).to.deep.equal(new ObjectId(objectId));
```
Comment (a-saad) [inline test/behaviorsSpec.js]: Let's go for a more compact comparison: 
expect(result[0]).to.deep.equal({_id: new ObjectId(objectId), $set: {name: 'crowdanalyzer'});

Comment (bahaagalal) [inline test/behaviorsSpec.js]: resolved

File: src/behaviors.js (hunk 1 above)
Comment (bahaagalal) [inline src/behaviors.js]: Resolved

File: src/behaviors.js (hunk 2 above)
Comment (bahaagalal) [inline src/behaviors.js]: resolved

**Pull request merged**